ta/ftpm/include/TEE/TpmToTEEMath.h

```diff
@@ -1,70 +1,72 @@
 /* SPDX-License-Identifier: BSD-2-Clause */
-/**********************************************************************
+/*
  * Copyright (c) 2024, Siemens AG
  * All rights reserved.
+ * Copyright (c) 2024, Linaro Limited
  *
  * Based on the original code by Microsoft. Modified to support using
  * TEE functions to provide cryptographic functionality.
  *
  * Portions Copyright Microsoft Corporation, see below for details:
  *
- *  The copyright in this software is being made available under the BSD License,
- *  included below. This software may be subject to other third party and
- *  contributor rights, including patent rights, and no such rights are granted
- *  under this license.
+ * The copyright in this software is being made available under the BSD
+ * License, included below. This software may be subject to other third
+ * party and contributor rights, including patent rights, and no such
+ * rights are granted under this license.
```

Comment:
Can we just drop the license text which is covered by `SPDX-*` here and other places? The copyrights should be retained though.

Reply:
I don't know if that's possible. The last time I checked, it deviated slightly from the license by `SPDX-License-Identifier: BSD-2-Clause`.

Reply:
IMO, we should have clarity regarding any deviation from standard BSD-2-Clause license with the authors. Otherwise it will be hard for users to understand licensing requirements.

Reply:
IANAL but for reasonable and practical purposes this is a BSD-2-Clause. I don't know what
```
 * The copyright in this software is being made available under the BSD
 * License, included below. This software may be subject to other third
 * party and contributor rights, including patent rights, and no such
 * rights are granted under this license.
```
is supposed to add. Perhaps it's only a lawyer trying to motivate its existence by causing troubles like this.
However, removing the license below affects the meaning of this text.

I propose that we keep this for now and if there's interest later we can try to remove it. fTPM has already been used with this license so we're not changing anything.

Reply:
Interesting and I see a corresponding patent already filed by Microsoft here: https://patents.google.com/patent/WO2013019369A1/en. IANAL too but this license text infers that there ins't any royalty free grant provided which is usually provided by a `BSD-2-Clause Plus Patent License` (https://spdx.org/licenses/BSD-2-Clause-Patent.html).

So I am not sure if bundling fTPM source within optee-os repo enforces that patent restriction on optee-os source/binary distribution as well. IMHO, we really need to seek lawyer's advice or consult with Microsoft. However, if we create a separate fTPM TA repo within OP-TEE github project then optee-os source may remain independent of any such restrictions.

Reply:
> Interesting and I see a corresponding patent already filed by Microsoft here: https://patents.google.com/patent/WO2013019369A1/en. IANAL too but this license text infers that there ins't any royalty free grant provided which is usually provided by a `BSD-2-Clause Plus Patent License` (https://spdx.org/licenses/BSD-2-Clause-Patent.html).

My interpretation of the quoted text is the opposite. They emphasize that they don't grant patent rights. But that's already implied by the `BSD-2-Clause` license, so the text is generally redundant.

> So I am not sure if bundling fTPM source within optee-os repo enforces that patent restriction on optee-os source/binary distribution as well. IMHO, we really need to seek lawyer's advice or consult with Microsoft. 

I'd say that it's not our problem.

> However, if we create a separate fTPM TA repo within OP-TEE github project then optee-os source may remain independent of any such restrictions.

It does not make a difference from a license point of view if files are in the same repository.

Reply:
> My interpretation of the quoted text is the opposite.

Lets leave that interpretation for lawyers or better consult with Microsoft for clarification. @jan-kiszka might have some clarity on this since they may be enabling fTPM in their products.

> > So I am not sure if bundling fTPM source within optee-os repo enforces that patent restriction on optee-os source/binary distribution as well. IMHO, we really need to seek lawyer's advice or consult with Microsoft.
> 
> I'd say that it's not our problem.

It will become a problem for downstream (like OE distro etc.) consumers of `optee-os` where just a version uprev will add a possible license restriction even if they aren't enabling fTPM. We at least need to update `optee-os` license accordingly such that it becomes known to consumers.

> > However, if we create a separate fTPM TA repo within OP-TEE github project then optee-os source may remain independent of any such restrictions.
> 
> It does not make a difference from a license point of view if files are in the same repository.

It will make a difference for consumers who choose to not enable fTPM in their products due to license constraints.

Reply:
>It will become a problem for downstream (like OE distro etc.) consumers of optee-os where just a version uprev will add a possible license restriction even if they aren't enabling fTPM. We at least need to update optee-os license accordingly such that it becomes known to consumers.

The main OP-TEE license is BSD-2-Clause, but there are exceptions. We don't have a uniform license that can be updated. We would in particular not want to add this since it only causes trouble when trying to classify the license we are just experiencing.

> > > However, if we create a separate fTPM TA repo within OP-TEE github project then optee-os source may remain independent of any such restrictions.
> > 
> > 
> > It does not make a difference from a license point of view if files are in the same repository.
> 
> It will make a difference for consumers who choose to not enable fTPM in their products due to license constraints.

Why? If the files aren't used why does it matter if they exist?

Reply:
I found an interesting read about "Patents in Open Source" here [1], please have a look at it. I hope it helps to clarify licensing concerns I am trying to raise here.

[1] https://google.github.io/opencasebook/patents/#express-but-non-specific-licenses